### PR TITLE
Create TodoList components and functionalities

### PR DESCRIPTION
**Pull request by NitBravoA92**
# 🎏 In this milestone, I:

done

✅ Create and configure a new REACTJS project using create-react-app

✅ Create the parent component `TodoApp` where the child components `Header` and `TodosLogic` are included

✅ Create the `Header` component where the title and subtitle of the application are displayed

✅ Create the `TodosLogic` component which is a container where the main state of the app is defined and includes the `InputTodo` and `TodosList` components

✅ Create the `InputTodo` component which has a form and an `onSubmit ` event to add a new task to the list

✅ Create the `TodosLots` component where the `TodoItem` component is imported and all tasks are listed

✅ Create the `TodoItem` component containing each task's JSX code. This code consists of a `<li>` element that contains an input `checkbox`, an `<h3>` element to display the title, and the edit and deletes buttons for the task. The delete button contains an `onClick()` event that triggers a handler function to delete a task. The input checkbox has an `onChange` event that triggers a handler function to update the status of a task.

✅ Add basic CSS styles for the ToDo List App.

# ⭐To the code reviewer 👨‍💻

💝 I really thank 🙇‍♂️ you, dear code reviewer 👨‍🎨 for dedicating your precious 🥇 time ⌚ to check the PR 🧾 .

Due to time ⌚ constraints, only the functionalities of Add task, delete task, and update task status have been created for this milestone. In the next Milestone, the Update task title functionality will be added.

I kindly ask 🙏 dear code reviewer 🤵 that if there is any kind of issue 🦯 in this project, please do list 📃 them in a descriptive 💡 manner and give your best suggestions 🎁 if needed.

### CODE REVIEW

File: src/components/Header.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import PropTypes from 'prop-types';
+
+const Header = ({ title, subtitle }) => (
+  <section id="header">
+    <h1>{ title }</h1>
+    <h2>{ subtitle }</h2>
+  </section>
+);
```
Comment (Gh-Reza) [inline src/components/Header.js]: Nice job so far 👏, I would suggest avoiding unnecessary complexity.  You can add the title and subtitle directly in the JSX code.

File: src/components/InputTodo.js [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      addTaskToState(taskTitle);
+      form.reset();
+    }
+  };
+
+  return (
+    <form id="form-input-todo" onSubmit={newTask}>
+      <div id="container-input-todo">
+        <input type="text" placeholder="Add todo..." id="inputDescription" />
+        <button type="submit" id="submit-new-task" aria-label="Add task"><FaCirclePlus /></button>
```
Comment (Gh-Reza) [inline src/components/InputTodo.js]: The input is not validated for an empty value. I kindly suggest you handle that using ```required``` attribute.

**Review by Gh-Reza — CHANGES_REQUESTED** [review]
Hi @NitBravoA92  , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- Project is working as expected 👍 
- Nice UI 👍
- Correct git flow 👍
- Add and delete functionality is working 👍

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Header.js (hunk 1 above)
Comment (NitBravoA92) [reply src/components/Header.js]: Hello @Gh-Reza Thank you very much for your review. I have taken your suggestions into account and have applied the suggested changes. You can see the changes in the last commits

File: src/components/InputTodo.js (hunk 2 above)
Comment (NitBravoA92) [reply src/components/InputTodo.js]: Hi again! @Gh-Reza Thank you so much for this review. Your recommendation is very correct. I have applied these changes, you can see them in the last commits.

**Review by Gh-Reza — APPROVED** [review]
Hi @NitBravoA92 , 

# STATUS: APPROVED ✔️ ✔️ 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations and well done! 🎉 
approved


## Highlights
- All linter checks are passing ✔️ ✔️ 
- Your PR is professional 👏 👏
- Required changes are applied 👍 👍  
- Meaning full commit messages ✨✨

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**